cpp/src/DataStorm/SessionI.cpp

```diff
@@ -655,20 +650,17 @@ SessionI::retry(NodePrx node, exception_ptr exception)
         }
     }
 
-    if (node->ice_getEndpoints().empty() && node->ice_getAdapterId().empty())
+    // Cancel any pending retry task, before we start a new one below.
+    if (_retryTask)
```

Comment:
Previous code only canceled the retry task when there is no endpoints. Here we are about to schedule a new task I think is clear to cancel any pending task.